src/main/java/uk/org/webcompere/modelassert/json/dsl/JsonNodeAssertDsl.java

```diff
@@ -161,6 +161,6 @@ default JsonAssertDslBuilders.At<A> at(String path) {
      * @return the {@link CoreJsonAssertion} for fluent assertions, with this condition added
      */
     default A isNotEmpty() {
-        return satisfies(not(new IsEmpty()));
+        return satisfies(not(MissingCondition.getInstance()).and(not(new IsEmpty())));
```

Comment:
It does the job for missing nodes, but what about `null` values ? It feels wrong that `assertJson("{\"prop\": null}").at("/prop").isNotEmpty();` passes.
```suggestion
        return satisfies(not(MissingCondition.getInstance())..and(not(NullCondition.getInstance())).and(not(new IsEmpty())));
```